### PR TITLE
feat: agent flow for proof presentation

### PR DESCRIPTION
**Pull request by berendsliedrecht**
Signed-off-by: Berend Sliedrecht <sliedrecht@berend.io>

### CODE REVIEW

File: example/src/credo/bleRequestProof.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
+import {
```
Comment (berendsliedrecht) [inline example/src/credo/bleRequestProof.ts:1]: This is the main request proof functionality

File: example/src/credo/bleShareProof.ts [hunk 2]
```diff
@@ -0,0 +1,200 @@
+import {
```
Comment (berendsliedrecht) [inline example/src/credo/bleShareProof.ts:1]: This is the main share proof functionality.

File: example/src/credo/BleProver.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  serviceUuid: string
+}
+
+export const BleProver: React.FC<BleProverProps> = ({ agent, serviceUuid }) => {
+  const [hasSharedProof, setHasSharedProof] = useState(false)
+  const { central } = useCentral()
+
+  const onFailure = () => console.error('[CENTRAL]: failure')
+  const onConnected = () => console.log('[CENTRAL]: connected')
+  const onDisconnected = () => console.log('[CENTRAL]: disconnected')
```
Comment (TimoGlastra) [inline example/src/credo/BleProver.tsx:21]: should we make the example a bit more in-depth? Would you usually do some reconnection stuff here?

File: example/src/credo/CredoScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+const uuid = 'd5477fcb-6e8b-4091-ad39-a1e30386ef76'
+
+export const CredoScreen = () => {
+  const [role, setRole] = useState<'prover' | 'verifier'>()
+  const [agent, setAgent] = useState<AppAgent>()
+
+  useEffect(() => {
+    setupAgent().then(setAgent)
+  }, [])
```
Comment (TimoGlastra) [inline example/src/credo/CredoScreen.tsx:21]: I've had issues with this in past, where setupAgent was called multiple times. I think we should handle it with state, or we can make it a `useSetupAgent` hook, which handles it with state internally ( and returns the agent afterwards ).

I know it's an example, but good to make the example follow best practices

File: example/src/credo/bleRequestProof.ts [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+        presentationTemplate.requestMessage,
+        serviceUuid
+      )
+      await peripheral.sendMessage(message)
+      connectedListener.remove()
+      resolve(proofRecordId)
+    })
+  })
+
+const disconnctedNotifier = (agent: AppAgent, peripheral: Peripheral, onDisconnected: () => Promise<void> | void) => {
```
Comment (TimoGlastra) [inline example/src/credo/bleRequestProof.ts:127]:
```suggestion
const disconnectedNotifier = (agent: AppAgent, peripheral: Peripheral, onDisconnected: () => Promise<void> | void) => {
```

File: example/src/credo/bleRequestProof.ts [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+          const { message } = await pp.acceptPresentation(agent.context, { proofRecord })
+          const serializedMessage = JsonTransformer.serialize(message)
+          await peripheral.sendMessage(serializedMessage)
+          off()
+          resolve(proofRecord)
+        } else if ([ProofState.Abandoned, ProofState.Declined].includes(proofRecord.state)) {
+          off()
+          reject(new Error(`Proof could not be shared because it has been ${proofRecord.state}`))
+        }
+      }
```
Comment (TimoGlastra) [inline example/src/credo/bleRequestProof.ts:160]: nit but usually `if (proofRecord.id !==) return` looks cleaner

File: example/src/credo/hooks/useBleRequestProof.tsx [hunk 7]
```diff
@@ -0,0 +1,109 @@
+// import {
```
Comment (TimoGlastra) [inline example/src/credo/hooks/useBleRequestProof.tsx:1]: not used anymore?

File: example/src/credo/utils/autoRespondToBleProofRequest.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        await agent.proofs.acceptRequest({ proofRecordId: proofRecord.id })
+
+        resolve(proofRecord)
+      } else if (proofRecord.state === ProofState.Done || proofRecord.state === ProofState.PresentationSent) {
+        const formatData = await agent.proofs.getFormatData(proofRecord.id)
+        const proofRepository = agent.dependencyManager.resolve(ProofRepository)
+        proofRecord.metadata.set(METADATA_KEY_FORMAT_DATA, formatData)
+        await proofRepository.update(agent.context, proofRecord)
+        agent.events.off(ProofEventTypes.ProofStateChanged, listener)
+        resolve(proofRecord)
```
Comment (TimoGlastra) [inline example/src/credo/utils/autoRespondToBleProofRequest.ts:32]: do we need this for the example?

File: patches/@hyperledger__indy-vdr-react-native.patch
Comment (TimoGlastra) [inline patches/@hyperledger__indy-vdr-react-native.patch:1]: I think we don't need two indy-vdr patches?

File: example/src/credo/BleProver.tsx (hunk 3 above)
Comment (berendsliedrecht) [reply example/src/credo/BleProver.tsx:24]: Really depends on the flow. I see them more as like UI update items. they are not too relevant.

File: example/src/credo/utils/autoRespondToBleProofRequest.ts (hunk 8 above)
Comment (berendsliedrecht) [reply example/src/credo/utils/autoRespondToBleProofRequest.ts]: No, not really.

**Review by TimoGlastra — APPROVED** [review]

**Review by TimoGlastra — APPROVED** [review]

**Pull request merged**